src/Calypso-Browser/ClyQueryViewMorph.class.st

```diff
@@ -433,12 +435,33 @@ ClyQueryViewMorph >> hasRealQuery [
 	^self query ~~ ClyUnknownQuery instance
 ]
 
+{ #category : #'drag and drop support' }
+ClyQueryViewMorph >> hasUserBeenFocusingOnItem: aSelectedItem [
+	targetDropItem ifNil: [ ^ false ].
+	hoverOverDropItemStartTime ifNil: [ ^ false].
+	aSelectedItem browserItem = targetDropItem browserItem ifFalse: [ ^ false ].
+	 ^ (hoverOverDropItemStartTime copy addSeconds: aSelectedItem class dragAndDropDelay / 1000) < Time now
```

Comment:
I would moved the dragAndDropDelay into ClyQueryViewMorh. And make it an instance of duration (2 seconds). In that case this code will be very simple and readable:
```Smalltalk
hoverOverDropItemStartTime + DragAndDropDelay < Time now
```

Reply:
Good point, thanks !

Reply:
Not so great for editing the setting though, it displays a duration.
So you have to know that it should be an integer... 
Although I like the idea, this does not really fit for user interaction, i think.
Still, I tried to put it anyway in #isReadyToExpand so the code is more readable.

Reply:
For settings you just need extra two accessors: 
```Smalltalk
ClyQueryViewMorph class>>dragAndDropDelaySeconds: aNumber
    DragAndDropDelay := aNumber seconds

ClyQueryViewMorph class>>dragAndDropDelaySeconds
    ^DragAndDropDelay asSeconds
```

Reply:
I feel seconds are not a good granularity.
Microseconds are too low.
Milliseconds seems ok, from a user perspective.

However there is a whole in the Duration/Time stuff and some methods are not available,
Microseconds, which is how I calculate the time is not, particularly.
Makes for harder calculation.

Anyway, that could be improved slightly, but I'm tired of this PR, so that;ll be good enough

Reply:
Good stuff anyway. 
I am just curious what was missing in the duration/time algebra and how we can improve it to make it suitable for such cases